pixl_core/README.md

```diff
@@ -161,3 +165,71 @@ so we can have different (or no) endpoints for different projects.
 For [testing](../test/README.md) we set up an additional Orthanc server that acts as a DICOMweb server,
 using the vanilla Orthanc Docker image with the DICOMWeb plugin enabled.
 
+## Uploading to an XNAT instance
+
+PIXL also supports sending DICOM images to an [XNAT](https://www.xnat.org/) instance.
+
+The `XNATUploader` class in `core.uploader._xnat` handles downloading anonymised images from Orthanc and
+sending to XNAT. [XNATPy](https://xnat.readthedocs.io/en/latest/) is used to upload the
+data to XNAT using the
+[`DICOM-zip` Import Handler](https://wiki.xnat.org/xnat-api/image-session-import-service-api#ImageSessionImportServiceAPI-SelectAnImportHandler).
+
+To use XNAT as an endpoint, first:
+
+- a user will need to be created on the XNAT instance to perform the upload with PIXL
+- a project will need to be created on the XNAT instance. It is assumed the user created for uploading
+  data does not have admin permissions to create new projects
+
+### XNAT endpoint Configuration
+
+Configuration for XNAT as an endpoint is done by storing the following secrets in an Azure Key Vault:
+
+```bash
+"${az_prefix}--xnat--host"  # hostname for the XNAT instance
+"${az_prefix}--xnat--username"  # username of user to perform upload
+"${az_prefix}--xnat--password"  # password of user to perform upload
+"${az_prefix}--xnat--port"  # port for connecting to the XNAT instance
+```
+
+where `az_prefix` is either the project slug or is defined in the [project configuration file](../template_config.yaml)
+as `azure_kv_alias`.
+
+> Note
+>
+> The project name defined in the configuration file **must** match the
+> [XNAT Project ID](https://wiki.xnat.org/documentation/creating-and-managing-projects). If the project name does
+> not match the XNAT Project ID, the upload will fail.
+
+The following environment variables must also be set to determine the XNAT destination and how to handle conflicts
+with existing session and series data:
+
+`"XNAT_DESTINATION"`:
+
+- if `"/archive"`, will send data straight to the archive
+- if `"/prearchive"`, will send data to the [prearchive](https://wiki.xnat.org/documentation/using-the-prearchive)
+  for manual review before archiving
+
+`"XNAT_OVERWRITE"`:
```

Comment:
yeah good point, we might want to configure them per project. Could this live in the project configuration file rather than azure, so it's easier to check and update the config for each project?

Reply:
After speaking with @HChughtai, perhaps having `XNAT_OVERWRITE=none` and `XNAT_DESTINATION=/archive` would be a good global default (and is the default XNAT behaviour). We could look into per-project configuration if any researchers request different settings?

Reply:
Ah yeah that sounds reasonable, let's thing about adding to config if we find a need for it later on

Reply:
yeah that sounds good, let's come back to it when the need arises